packages/modules/viewer-react/src/components/iModel/IModelLoader.tsx

```diff
@@ -19,182 +19,152 @@ import {
   useTheme,
   useUiProviders,
 } from "../../hooks";
+import { getAndSetViewState } from "../../services/iModel";
 import {
-  getAndSetViewState,
-  openLocalIModel,
-  openRemoteIModel,
-} from "../../services/iModel";
+  gatherRequiredViewerProps,
+  openConnection,
+} from "../../services/iModel/iModelViewerHelper";
 import { ViewerPerformance } from "../../services/telemetry";
-import type {
-  BlankViewerProps,
-  ConnectedViewerProps,
-  FileViewerProps,
-  LoaderProps,
-} from "../../types";
+import type { ModelLoaderProps } from "../../types";
 import { IModelViewer } from "./IModelViewer";
 
-type ModelLoaderProps = Partial<
-  ConnectedViewerProps & FileViewerProps & BlankViewerProps
-> &
-  LoaderProps;
-
-const IModelLoader = React.memo(
-  ({
-    iModelId,
-    iTwinId,
-    changeSetId,
-    onIModelConnected,
-    filePath,
+const IModelLoader = React.memo((viewerProps: ModelLoaderProps) => {
+  const [error, setError] = useState<Error>();
+  const [connection, setConnection] = useState<IModelConnection>();
+  const isMounted = useIsMounted();
+  const {
     frontstages,
     defaultUiConfig,
-    backstageItems,
     viewportOptions,
-    blankConnection,
+    viewCreatorOptions,
     blankConnectionViewState,
     uiProviders,
     theme,
-    viewCreatorOptions,
+    onIModelConnected,
+    backstageItems,
     loadingComponent,
-  }: ModelLoaderProps) => {
-    const [error, setError] = useState<Error>();
-    const [connection, setConnection] = useState<IModelConnection>();
-    const isMounted = useIsMounted();
-    const { finalFrontstages, noConnectionRequired, customDefaultFrontstage } =
-      useFrontstages({
-        frontstages,
-        defaultUiConfig,
-        viewportOptions,
-        viewCreatorOptions,
-        blankConnectionViewState,
-      });
-
-    useUiProviders(uiProviders);
-    useTheme(theme);
-
-    const getModelConnection = useCallback(async (): Promise<
-      IModelConnection | undefined
-    > => {
-      if (!(iTwinId && iModelId) && !filePath && !blankConnection) {
-        throw new Error(
-          IModelApp.localization.getLocalizedStringWithNamespace(
-            "iTwinViewer",
-            "missingConnectionProps"
-          )
-        );
-      }
+  } = viewerProps;
 
-      ViewerPerformance.addMark("IModelConnectionStarted");
-      ViewerPerformance.addMeasure(
-        "IModelConnecting",
-        "ViewerStarting",
-        "IModelConnectionStarted"
+  const { finalFrontstages, noConnectionRequired, customDefaultFrontstage } =
+    useFrontstages({
+      frontstages,
+      defaultUiConfig,
+      viewportOptions,
+      viewCreatorOptions,
+      blankConnectionViewState,
+    });
+  useUiProviders(uiProviders);
+  useTheme(theme);
+
+  const getModelConnection = useCallback(async (): Promise<
+    IModelConnection | undefined
+  > => {
+    const sanitizedProps = gatherRequiredViewerProps(viewerProps);
```

Comment:
nit: rename to requiredProps or requiredConnectionProps or something?

Reply:
done